### PR TITLE
Stack id

### PR DESCRIPTION
**Pull request by Cool04ek**

### CODE REVIEW

File: lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java [hunk 1]
```diff
@@ -21,15 +23,23 @@
...
+    public TopBarAnimator(TopBar topBar, @Nullable String stackId) {
+        this.topBar = topBar;
+        this.stackId = stackId;
+    }
+
     public void show(AnimationOptions options) {
-        if (options.hasValue()) {
+        if (options.hasValue() && options.id.hasValue() && options.id.get().equals(stackId)) {
+            showAnimator = options.getAnimation(topBar);
+        } else if (options.hasValue() && !options.id.hasValue()) {
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java]: Isn't `options.id.hasValue()` always `false` here?

File: lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java [hunk 2]
```diff
@@ -21,15 +23,23 @@
...
 
+    public TopBarAnimator(TopBar topBar, @Nullable String stackId) {
+        this.topBar = topBar;
+        this.stackId = stackId;
+    }
+
     public void show(AnimationOptions options) {
-        if (options.hasValue()) {
+        if (options.hasValue() && options.id.hasValue() && options.id.get().equals(stackId)) {
+            showAnimator = options.getAnimation(topBar);
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java]: Line 41 equals line 43... is this conditional needed?

File: lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java [hunk 3]
```diff
@@ -62,7 +72,9 @@ private AnimatorSet getDefaultShowAnimator(float startTranslation, TimeInterpola
     }
 
     public void hide(AnimationOptions options, AnimationListener listener) {
-        if (options.hasValue()) {
+        if (options.hasValue() && options.id.hasValue() && options.id.get().equals(stackId)) {
+            hideAnimator = options.getAnimation(topBar);
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java]: line 76 is the same as line 78, is this conditional required?

File: lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java (hunk 2 above)
Comment (Cool04ek) [reply lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java]: No. The difference is, that first condition is true, if you have provided id and it's the same. And second condition true if you didn't provide id at all

File: lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java (hunk 3 above)
Comment (Cool04ek) [reply lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java]: Same here

**Pull request merged**